### PR TITLE
鸿蒙平台适配

### PR DESCRIPTION
**Pull request by IceBBBBB**
鸿蒙平台适配

### CODE REVIEW

File: tests/lua-tests/project/proj.ohos/entry/src/main/ets/pages/Index.ets [hunk 1]
```diff
@@ -68,10 +73,38 @@ struct Index {
...
+        message: "确认退出游戏吗",
+        buttons: [
+          {
+            text: '取消',
+            // 颜色可自定义
+            color: '#000000'
+          },
+          {
+            text: '确认',
+            // 颜色可自定义
```
Comment (minggo) [inline tests/lua-tests/project/proj.ohos/entry/src/main/ets/pages/Index.ets]: It's better not use Chinese.

File: cocos/2d/CCDirector.h [hunk 2]
```diff
@@ -384,6 +384,9 @@ class CC_DLL Director : public Ref
     float getFrameRate() const { return _frameRate; }
 
 protected:
+    #if  (CC_TARGET_PLATFORM == CC_PLATFORM_OHOS)
+        friend class GLView;
+    #endif
```
Comment (minggo) [inline cocos/2d/CCDirector.h]: Why have to do like this?

File: cocos/2d/platform/ohos/napi/modules/InputNapi.cpp [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    
+    size_t textLen;
+    char text[2560] = {0};
+    NAPI_CALL(env, napi_get_value_string_utf8(env, args[0], text, 2560, &textLen));
+
+    // 优化: 使用string_view避免不必要的字符串拷贝
+    std::string_view oldView(oldContent);
+    std::string_view newView(text, textLen);
+
+    // 找到第一个不同的字符位置
```
Comment (minggo) [inline cocos/2d/platform/ohos/napi/modules/InputNapi.cpp]: It is better not use Chinese

**Comment by minggo** [comment]
Why need opengles header files in external?

File: tests/lua-tests/project/proj.ohos/entry/src/main/ets/pages/Index.ets (hunk 1 above)
Comment (wanghui187) [reply tests/lua-tests/project/proj.ohos/entry/src/main/ets/pages/Index.ets]: fix

File: cocos/2d/CCDirector.h (hunk 2 above)
Comment (wanghui187) [reply cocos/2d/CCDirector.h]: Deleted

File: cocos/2d/platform/ohos/napi/modules/InputNapi.cpp (hunk 3 above)
Comment (wanghui187) [reply cocos/2d/platform/ohos/napi/modules/InputNapi.cpp]: fix

**Comment by minggo** [comment]
> Why need opengles header files in external?

@IceBBBBB why need it?

**Pull request merged**